source/dub/commandline.d

```diff
@@ -1181,8 +1182,16 @@ class GenerateCommand : PackageBuildCommand {
 		gensettings.single = m_single;
 
 		logDiagnostic("Generating using %s", m_generator);
-		dub.generateProject(m_generator, gensettings);
+
+		// With an explicitly requested `unittest` config, switch to the special test
+		// runner config (which doesn't require an existing `unittest` configuration).
+		if (m_buildConfig == "unittest")
+			dub.testProject(m_generator, gensettings, null, NativePath());
+		else
```

Comment:
this should not be special to the command-line, I think this makes sense to have for API users as well (move to dub.d)

Reply:
I was thinking about it, but then chose the safer path, as `generateProject()` is called in some more places. I wasn't thinking about dub-library users at all.